content/sensu-go/6.5/observability-pipeline/observe-process/tcp-stream-handlers.md

```diff
@@ -0,0 +1,496 @@
+---
+title: "TCP stream handlers reference"
+linkTitle: "TCP Stream Handlers Reference"
+reference_title: "TCP stream handlers"
+type: "reference"
+description: "TCP stream handlers are actions the Sensu backend executes on events, allowing you to create automated monitoring workflows. Read the reference doc to learn about TCP stream handlers."
+weight: 17
+version: "6.5"
+product: "Sensu Go"
+platformContent: false
+menu:
+  sensu-go-6.5:
+    parent: observe-process
+---
+
+{{% notice commercial %}}
+**COMMERCIAL FEATURE**: Access TCP stream handlers in the packaged Sensu Go distribution.
+For more information, read [Get started with commercial features](../../commercial/).
+{{% /notice %}}
+
+Sensu executes TCP stream handlers during the **[process][22]** stage of the [observability pipeline][29].
+
+Like traditional [TCP handlers][1], TCP stream handlers send observability event data to TCP sockets for external services to consume.
+However, TCP stream handlers can help prevent the data bottlenecks you may experience with traditional TCP handlers.
+
+Traditional TCP handlers start a new UNIX process for every Sensu event they receive and require a new connection to send every event.
+As you scale up and process more events per second, the rate at which the TCP handler can transmit observability event data decreases.
+
+TCP stream handlers allow you to configure a connection pool with a minimum and maximum number of connections for the handler to use.
+For example, suppose you configure a TCP stream handler with a pool of 5 to 10 connections, and 1000 events are queued for transmission.
+As each connection finishes transmitting an event, it becomes available again and returns to the pool so the handler can use it to send the next event in the queue.
+
+TCP stream handlers will reuse the available connections as long as they can rather than requiring a new connection for every event, which increases event throughput.
+In addition to providing a persistent TCP connection to transmit Sensu observation events to a remote data storage service, TCP stream handlers allow you to use transport layer security (TLS) for secure data transmission.
+
+TCP stream handlers are commercial resources available for use in [pipeline definitions][2].
+
+## TCP stream handler example
+
+This example shows a TCP stream handler resource definition configured to use TLS:
+
+{{< language-toggle >}}
+
+{{< code yml >}}
+---
+type: TCPStreamHandler
+api_version: pipeline/v1
+metadata:
+  name: logstash
+  namespace: default
+spec:
+  address: 127.0.0.1:4242
+  tls_ca_cert_file: "/path/to/tls/ca.pem"
+  tls_cert_file: "/path/to/tls/cert.pem"
+  tls_key_file: "/path/to/tls/key.pem"
+  max_connections: 10
+  min_connections: 5
```

Comment:
This field does not exist in the implementation. It did exist in the initial spec, but was removed after some consideration.